src/Core/Http/SecureHttpClient.php

```diff
@@ -136,17 +138,77 @@ public function signedRequest($method, $endpoint, array $payload = [], $returnJs
         }
     }
 
+    private function getAlg()
+    {
+        $privateKey = $this->accountKeyPair->getPrivateKey();
+        $parsedKey = $this->keyParser->parse($privateKey);
+        switch ($parsedKey->getType()) {
+            case OPENSSL_KEYTYPE_RSA:
+                return 'RS256';
+            case OPENSSL_KEYTYPE_EC:
+                switch ($parsedKey->getBits()) {
+                    case 256:
+                    case 384:
+                        return 'ES'.$parsedKey->getBits();
+                    case 521:
```

Comment:
512?

Reply:
nop... https://tools.ietf.org/html/rfc4754#section-4 probably a math logic behind this

Reply:
Okay :) !